modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/model/extension/xml/ComponentModelReaderHelper.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.config.internal.dsl.model.extension.xml;
+
+import org.mule.runtime.config.internal.model.ComponentModel;
+import org.mule.runtime.core.privileged.execution.LocationExecutionContextProvider;
+
+import static java.lang.System.lineSeparator;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static org.apache.commons.lang3.StringUtils.repeat;
+import static org.mule.runtime.core.privileged.execution.LocationExecutionContextProvider.maskPasswords;
+
+/**
+ * Internal object responsible to traverse the entire XMl of a given application to read the {@link ComponentModel} to
+ * recreate the macro expanded XML of the application.
+ *
+ * It will obfuscate attributes that matches with {@link LocationExecutionContextProvider#maskPasswords(String, String)}.
+ * @since 4.2.0
+ */
+class ComponentModelReaderHelper {
+
+  static final String PASSWORD_MASK = "@@credentials@@";
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: 'PASSWORD' detected in this expression, review this potentially hard-coded credential. ([squid:S2068](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2068))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7659&issues=AWjN2sq4otIXnmsUgbPf&open=AWjN2sq4otIXnmsUgbPf)

[ISSUE_KEY]: AWjN2sq4otIXnmsUgbPf